src/main/java/com/meta/cp4m/metrics/MetricsCollector.java

```diff
@@ -0,0 +1,22 @@
+/*
+ *
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.meta.cp4m.metrics;
+
+import com.meta.cp4m.llm.OpenAIPlugin;
+import com.meta.cp4m.message.MessageHandler;
+
+import java.util.UUID;
+
+public interface MetricsCollector {
+
+    <T extends MessageHandler<?>, E extends OpenAIPlugin<?>> void logMessageReceived(String name, UUID instanceId, Class<T> messengerType, Class<E> pluginType, String appId, String businessContactIdentifier);
+
+    <T extends MessageHandler<?>, E extends OpenAIPlugin<?>> void logMessageSent(String name, UUID instanceId, Class<T> messengerType, Class<E> pluginType, String appId, String businessContactIdentifier);
```

Comment:
you don't need the `name` in either of these methods, the name of the event is in the method name ie message received & message sent

Reply:
You also don't need instance id, that will be the instance of the cp4m instance. For the time being just create a random uuid in a private static final variable and use it